source/extensions/common/wasm/context.cc

```diff
@@ -1662,10 +1665,15 @@ WasmResult Context::sendLocalResponse(uint32_t response_code, std::string_view b
                                   modify_headers = std::move(modify_headers), grpc_status,
                                   details = StringUtil::replaceAllEmptySpace(
                                       absl::string_view(details.data(), details.size()))] {
+      if (local_reply_sent_) {
+        return;
+      }
       decoder_callbacks_->sendLocalReply(static_cast<Envoy::Http::Code>(response_code), body_text,
                                          modify_headers, grpc_status, details);
+      local_reply_sent_ = true;
     });
   }
+  local_reply_hold_ = true;
```

Comment:
Why do we need both hold and sent? Doesn't your "hold" case already cover the sent case?

Reply:
https://github.com/envoyproxy/envoy/blob/f09ed3646b88fae777a060be37f3224dfa67f428/source/extensions/common/wasm/context.cc#L1609

https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/4fcf895fa2433a1cdf20704926b8b7e4039a6a04/src/context.cc#L34

This is to avoid sending the local reply twice if the vm fails. I added another test for this case.

Reply:
I'll defer to @mathetake @lizan here, can you folks take a pass as CODEOWNERS? Thanks.

Reply:
yeah unfortunately I think it is inevitable to have both of these two booleans.

Reply:
@johnlanni this is fine I think, can you add some documentation in the form of comments for posterity, as the reasons we need this is subtle as per your above comment? Everything else LGTM.